multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2020-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package org.jetbrains.kotlinx.multik.cuda
+
+import jcuda.Pointer
+import jcuda.jcublas.JCublas
+import org.jetbrains.kotlinx.multik.api.LinAlg
+import org.jetbrains.kotlinx.multik.ndarray.data.*
+
+public object CudaLinAlg : LinAlg {
+    init {
+        CudaEngine
+    }
+
+    override fun <T : Number> pow(mat: MultiArray<T, D2>, n: Int): NDArray<T, D2> {
+        TODO("Not yet implemented")
+    }
+
+    override fun <T : Number> norm(mat: MultiArray<T, D2>, p: Int): Double {
+        TODO("Not yet implemented")
+    }
+
+    override fun <T : Number> inv(mat: MultiArray<T, D2>): NDArray<T, D2> {
+        TODO("Not yet implemented")
+    }
+
+    override fun <T : Number, D : Dim2> solve(a: MultiArray<T, D2>, b: MultiArray<T, D>): NDArray<T, D> {
+        TODO("Not yet implemented")
+    }
+
+    private fun <T : Number, D: Dim2> isTransposedConsistent(x: MultiArray<T, D>): Boolean {
+        return x.transpose().consistent
+    }
+
+    private fun <T : Number, D: Dim2> getConsistentOrTransposedConsistent(x: MultiArray<T, D>): Pair<MultiArray<T, D>, Boolean> {
+        return when {
+            x.consistent -> x to false
+            x.dim.d == 2 && isTransposedConsistent(x) -> x to true
```

Comment:
Why is the condition `x.dim.d == 2` needed here?

Reply:
If MultiArray is a vector (x.dim.d != 2) then the following condition is always false